python/cudf/cudf/core/column/column.py

```diff
@@ -1240,6 +1240,11 @@ def as_column(arbitrary, nan_as_null=None, dtype=None, length=None):
                 size=pa_size,
                 offset=pa_offset,
             )
+        elif isinstance(arbitrary, pa.ListArray):
+            raise ValueError(
+                "cudf Column doesn't support list like data types"
+            )
```

Comment:
Let's use `NotImplementedError` as this is something we don't yet support, as opposed to a problem with user-provided data. Also let's avoid using the term `Column` in public errors. 

```suggestion
            raise NotImplementedErr(
                "cudf doesn't support list like data types"
            )
```